File: soundboard.cpp
#include <iostream>
#include <thread>
#include "include/raylib.h"
#include <vector>
using namespace std;
#define vec vector
#define str string

const vec<str> sound_files = {
    "sound1.wav", "sound2.wav", "sound3.wav", "sound4.wav",
    "sound5.wav", "sound6.wav", "sound7.wav", "sound8.wav", "sound9.wav"
};

int main(void) {
    const int screenWidth = 800;
    const int screenHeight = 450;
    InitWindow(screenWidth, screenHeight, "C Sound Board");
    SetTargetFPS(60);

    InitAudioDevice(); 

    Sound fxWav = LoadSound("resources/sound.wav");  
    Sound fxOgg = LoadSound("resources/target.ogg"); 

    
    while (!WindowShouldClose()) {
        if (IsKeyPressed(KEY_SPACE))
            PlaySound(fxWav);
        if (IsKeyPressed(KEY_ENTER))
            PlaySound(fxOgg);
        
        BeginDrawing();
        ClearBackground(RAYWHITE);
        DrawText("Press SPACE to PLAY the WAV sound!", 200, 180, 20, LIGHTGRAY);
        DrawText("Press ENTER to PLAY the OGG sound!", 200, 220, 20, LIGHTGRAY);
        EndDrawing();
        
    }


    UnloadSound(fxWav); 
    UnloadSound(fxOgg); 
    CloseAudioDevice(); 
    CloseWindow(); 
    return 0;
}

File: soundboard.py
import pygame
from pygame.locals import K_1, K_2, K_3, K_4, K_5, K_6, K_7, K_8, K_9

# Initialize pygame mixer and window
pygame.init()
pygame.mixer.init()

# Load sounds for keys 1-9
sounds = {
    K_1: pygame.mixer.Sound('sound1.wav'),
    K_2: pygame.mixer.Sound('sound2.wav'),
    K_3: pygame.mixer.Sound('sound3.wav'),
    K_4: pygame.mixer.Sound('sound4.wav'),
    K_5: pygame.mixer.Sound('sound5.wav'),
    K_6: pygame.mixer.Sound('sound6.wav'),
    K_7: pygame.mixer.Sound('sound7.wav'),
    K_8: pygame.mixer.Sound('sound8.wav'),
    K_9: pygame.mixer.Sound('sound9.wav'),
}

# Main loop
running = True
screen = pygame.display.set_mode((200, 200))
pygame.display.set_caption("Soundboard")

while running:
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        if event.type == pygame.KEYDOWN and event.key in sounds:
            sounds[event.key].play()

pygame.quit()
